privateFields_test.go

```diff
@@ -0,0 +1,28 @@
+package protobuf
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+type Private struct {
+	a int
+}
+
+type Public struct {
+	A int
+}
+
+func TestPrivate(t *testing.T) {
+	s := Private{37}
+	u := Public{37}
+
+	bufS, errS := Encode(&s)
+	bufU, errU := Encode(&u)
```

Comment:
Also try
```
bufT, errT := Encode(&Public{})
require.Nil(t, errT)
```

Reply:
So in the `&Public{}` case it still outputs `[8 0] <nil>` when I print buf and err.
And you're right, in the pointer to an empty string case, the if statement I added outputs an error.

The only other option I see is the alternative proposed earlier:
"""
The alternative I can see us doing is:
in (en *encoder) message, if no field.CanSet() was true in the loop, we can do something. However, similarly to the issue #30 no other method than Encode can return errors.
"""